protocol/protocol.re

```diff
@@ -147,6 +147,10 @@ let apply_block = (state, block) => {
         block.state_root_hash != state.state_root_hash
           ? Unix.time() : state.last_state_root_update,
       last_applied_block_timestamp: Unix.time(),
+      past_state_root_hashes: [
+        state.state_root_hash,
```

Comment:
This should only be added if the state_root_hash is different than the previous one